common/test/acceptance/tests/lms/test_lms_gating.py

```diff
@@ -204,8 +208,68 @@ def test_gated_subsection_in_lms_for_staff(self):
         self.course_home_page.visit()
         self.course_home_page.preview.set_staff_view_mode('Learner')
         self.course_home_page.wait_for_page()
-        self.assertEqual(self.course_home_page.outline.num_subsections, 2)
+        self.assertEqual(self.course_home_page.outline.num_subsections, 3)
         self.course_home_page.outline.go_to_section('Test Section 1', 'Test Subsection 1')
         self.courseware_page.wait_for_page()
         # banner displayed informing section is a prereq
         self.assertTrue(self.courseware_page.has_banner())
+
+    def test_gating_banner_before_special_exam(self):
```

Comment:
👍 this unit test is very readable but can we rename the test to be more readable? I am good with it but just wondering if we can have more readable name.

Thank you for your work on this.

Reply:
Thank you @DawoudSheraz for adding the test.